core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/TrackNetworkInfra.kt

```diff
@@ -27,12 +27,14 @@ typealias TrackNodePortId = StaticIdx<TrackNodePort>
 sealed interface TrackNodeConfig
 typealias TrackNodeConfigId = StaticIdx<TrackNodeConfig>
 
-
+@Suppress("INAPPLICABLE_JVM_NAME")
 interface TrackNetworkInfra {
-    fun getNextTrackSection(current: DirTrackSectionId, config: TrackNodeConfigId): OptDirTrackSectionId
+    fun getNextTrackSection(current: EndpointTrackSectionId, config: TrackNodeConfigId): OptDirTrackSectionId
+    @JvmName("getNextTrackSections")
+    fun getNextTrackSections(trackEndpoint: EndpointTrackSectionId): DirStaticIdxList<TrackSection>
```

Comment:
I would like to point out that this API can be emulated using an extension function: you can get all possible following track sections by enumerating all possible track node configs, and calling getNextTrackSection for each one.

Doing so has the upside of providing both switch configurations and next track sections at once.

It's also only used for a sanity check, so I'm all for keeping this code if makes things easier, but would rather do it outside of the main API.